pyctcdecode/language_model.py

```diff
@@ -271,6 +275,71 @@ def score(
         lm_score = self.alpha * lm_score * LOG_BASE_CHANGE_FACTOR + self.beta
         return lm_score, end_state
 
+    @classmethod
+    def load_from_hf_hub(
+        cls,
+        pretrained_path: str,
+        unigrams: Optional[Collection[str]] = None,
+        alpha: float = DEFAULT_ALPHA,
+        beta: float = DEFAULT_BETA,
+        unk_score_offset: float = DEFAULT_UNK_LOGP_OFFSET,
+        score_boundary: bool = DEFAULT_SCORE_LM_BOUNDARY,
+        model_file_name: str = DEFAULT_MODEL_FILE_NAME,
+    ):
+        """Class method to load model from https://huggingface.co/
+
+        Args:
+            pretrained_path: string, the `model id` of a pretrained model hosted inside a model
+                repo on https://huggingface.co/. Valid model ids can be namespaced under a user or
+                organization name, like ``kensho/5gram-spanish-kenLM``. For more information, please
+                take a look at https://huggingface.co/docs/hub/main .
+            unigrams: list of known word unigrams
+            alpha: weight for language model during shallow fusion
+            beta: weight for length score adjustment of during scoring
+            unk_score_offset: amount of log score offset for unknown tokens
+            score_boundary: whether to have kenlm respect boundaries when scoring
+            model_file_name: file name of the model as expected to be saved on https://huggingface.co/.
+        """
+        if os.path.isfile(pretrained_path):
+            logger.info(f"Path to local file {pretrained_path} provided. Loading KenLM model from local file.")
+            kenlm_model_path = pretrained_path
+        else:
+            from . import __version__ as VERSION, __package_name__ as LIBRARY_NAME
+
+            CACHE_DIRECTORY = os.path.join(Path.home(), ".cache", LIBRARY_NAME)
+
+            try:
+                from huggingface_hub import hf_hub_download
+            except ImportError:
+                raise ImportError(
+                    "You need to install huggingface_hub to use `load_from_hf_hub`. "
+                    "See https://pypi.org/project/huggingface-hub/ for installation."
+                )
+
+            # download and cache model
+            kenlm_model_path = hf_hub_download(
+                repo_id=pretrained_path, 
+                filename=model_file_name,
+                library_name=LIBRARY_NAME,
+                library_version=VERSION,
+                cache_dir=CACHE_DIRECTORY,
+            )
+
+        # load unigrams if possible
+        if unigrams is None:
+            try:
+                unigrams = load_unigram_set_from_arpa(kenlm_model_path)
+            except ValueError:
+                logger.warning(
+                    "Unigrams not provided and cannot be automatically determined from LM file (only "
+                    "arpa format). Decoding accuracy might be reduced."
+                )
```

Comment:
I don't love that we are re-writing a lot of the functionality of `build_ctcdecoder` here, did we choose the abstraction right? An alternative for example could be that instead of having a method on the LM class, we write a helper that downloads the model and then provides a local path as output that can be used like any other model in `build_ctcdecoder`. That would help minimize code duplication